src/common/k8s-api/kube-api.ts

```diff
@@ -261,39 +308,76 @@ export interface DeleteResourceDescriptor extends ResourceDescriptor {
   propagationPolicy?: PropagationPolicy;
 }
 
-export class KubeApi<T extends KubeObject> {
+/**
+ * @deprecated In the new extension API, don't expose `KubeApi`'s constructor
+ */
+function legacyRegisterApi(api: KubeApi<any, any>): void {
+  // Try both just in case, because we might be in a testing environment
```

Comment:
This isn't working properly in unit tests. When called from renderer, it will emit the event in renderer and main process which isn't the case in production. LegacyGlobals are not welcome to unit tests because of the issue that state is shared.

Reply:
This conversation was resolved without message for how or if it was fixed.

Reply:
Talked with Janne, resolved.